constraint/system.go

```diff
@@ -46,7 +46,8 @@ type ConstraintSystem interface {
 
 	// AddSolverHint adds a hint to the solver such that the output variables will be computed
 	// using a call to output := f(input...) at solve time.
-	AddSolverHint(f solver.Hint, input []LinearExpression, nbOutput int) (internalVariables []int, err error)
+	// Providing the function f is optional.
+	AddSolverHint(f solver.Hint, id solver.HintID, input []LinearExpression, nbOutput int) (internalVariables []int, err error)
```

Comment:
"Providing the function f is optional." --> describe a bit more what happens (ie if f == nil, then ... )